train.py

```diff
@@ -165,17 +178,30 @@ def main():
         else:
             param.update_rule.add_hook(WeightDecay(0.0005))
 
-    updater = training.updaters.StandardUpdater(
-        train_iter, optimizer, device=args.gpu)
+    if len(args.gpu) <= 1:
+        updater = training.updaters.StandardUpdater(
+            train_iter[0],
+            optimizer,
+            device=args.gpu[0] if len(args.gpu) > 0 else -1,
+        )
+    else:
+        updater = training.updaters.MultiprocessParallelUpdater(
+            train_iter, optimizer, devices=args.gpu)
+        updater.setup_workers()
+
     trainer = training.Trainer(updater, (120000, 'iteration'), args.out)
 
+    # trainer.extend(
+    #     extensions.ExponentialShift('lr', 0.1, init=1e-3),
+    #     trigger=triggers.ManualScheduleTrigger([80000, 100000], 'iteration'))
```

Comment:
Is this something we might need in the future?

Reply:
Hmm.... I'm not sure... maybe but we can also kill it